docs/src/pages/components/data-grid/components/components.md

```diff
@@ -7,10 +7,83 @@ components: DataGrid, XGrid
 
 <p class="description">The grid is highly customizable. Override components using the <code>components</code> prop.</p>
 
+## Installing Component
+
 As part of the customization API, the grid allows you to replace and override nested components with the `components` prop.
-The prop accepts an object of type [GridSlotsComponent](/api/data-grid/#slots) .
+The prop accepts an object of type [`GridSlotsComponent`](/api/data-grid/#slots).
+
+If you wish to pass some props in a custom component, you can do it using the `componentsProps` prop. This prop is of type `GridSlotsComponentsProps`.
+
+As an example, you could override the column menu with some props as below.
+
+```jsx
+<DataGrid
+  rows={rows}
+  columns={columns}
+  components={{
+    ColumnMenu: MyCustomColumnMenu,
+  }}
+  componentsProps={{
+    columnMenu: { background: 'red', counter: rows.length },
+  }}
+/>
+```
+
+**Note** The casing is different between the `components` and `componentsProps` prop.
+
+### Getting props
+
+While building new component slots, you might need some grid data.
+Therefore, we exposed a `useGridSlotComponentProps` hook which allows to retrieve the following props.
+
+- `state`: the current grid state.
+- `rows`: the current rows in the grid.
+- `columns`: the current columns in the grid.
+- `options`: the current set of options in the grid.
+- `apiRef`: the `GridApi` ref that allows to manipulate the grid.
```

Comment:
I would add a pro icon as it's not in the MIT version

Reply:
atm params are not segregated for data-grid

Reply:
If we plan to do such, it would be much better to document it now. Why? I think that developers would hate us and we would break trust if we move features from the community plan to the pro plan.

Reply:
done